packages/sdk/src/types/internal/actor.ts

```diff
@@ -64,8 +65,8 @@ export class InternalActor implements InternalPatchable<ActorLike> {
         if (patch) {
             patch.id = this.actor.id;
             delete this.patch;
+            return Actor.sanitize(patch);

```

Comment:
>return Actor.sanitize(patch); [](start = 12, length = 29)

This isn't the right place to resolve json values. We can't have unresolved objects in the patch between frames or they might get updated before serialization.

Reply:
This is really the only place to do it. We can't resolve every time the patch updates, it would produce a ton of garbage objects that get immediately discarded. Is it really so bad if they change? Which objects end up in the patch that don't trigger updates on changed anyway?

---
In reply to: [268830814](https://github.com/Microsoft/mixed-reality-extension-sdk/pull/274#discussion_r268830814) [](ancestors = 268830814)